Eclipse Orbit's 2023-09 M3 broke CDT

There is a failure to build that looks like it may be due to a change in our dependencies:

```
Error: Failed to resolve target definition file:/home/runner/work/cdt/cdt/releng/org.eclipse.cdt.target/cdt.target: Could not find "javax.inject/0.0.0" in the repositories of the current location -> [Help 1]
Error:
Error: To see the full stack trace of the errors, re-run Maven with the -e switch.
Error: Re-run Maven using the -X switch to enable full debug logging.
Error:
Error: For more information about the errors and possible solutions, please read the following articles:
Error: [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionException
Error: Process completed with exit code 1.

```

I am also looking at that now - it is probably related to #495

--- a/debug/org.eclipse.cdt.debug.application.product/debug.product
+++ b/debug/org.eclipse.cdt.debug.application.product/debug.product
@@ -190,10 +190,10 @@ Java and all Java-based trademarks are trademarks of Oracle Corporation in the U
       <plugin id="com.sun.xml.bind"/>
       <plugin id="jakarta.activation-api"/>
       <plugin id="jakarta.annotation-api"/>
+      <plugin id="jakarta.inject.jakarta.inject-api"/>
       <plugin id="jakarta.servlet-api"/>
       <plugin id="jakarta.xml.bind-api"/>
       <plugin id="javax.el-api"/>
-      <plugin id="javax.inject"/>
       <plugin id="javax.servlet.jsp-api"/>
       <plugin id="org.apache.aries.spifly.dynamic.bundle"/>
       <plugin id="org.apache.batik.constants"/>

--- a/releng/org.eclipse.cdt.target/cdt.target
+++ b/releng/org.eclipse.cdt.target/cdt.target
@@ -1,13 +1,13 @@
 <?xml version="1.0" encoding="UTF-8" standalone="no"?>
 <?pde version="3.8"?>
-<target name="cdt" sequenceNumber="143">
+<target name="cdt" sequenceNumber="144">
 	<locations>
 		<location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="planner" includeSource="true" type="InstallableUnit">
 			<repository location="https://download.eclipse.org/cbi/updates/license/" />
 			<unit id="org.eclipse.license.feature.group" version="0.0.0" />
 		</location>
 		<location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="planner" includeSource="true" type="InstallableUnit">
-			<repository location="https://download.eclipse.org/eclipse/updates/4.29-I-builds/I20230705-1800/" />
+			<repository location="https://download.eclipse.org/eclipse/updates/4.29-I-builds/I20230815-0600/" />
 			<unit id="org.eclipse.equinox.executable.feature.group" version="0.0.0" />
 			<unit id="org.eclipse.jdt.annotation" version="0.0.0" />
 			<unit id="org.eclipse.sdk.feature.group" version="0.0.0" />
@@ -194,10 +194,9 @@
 			This dep can't be added to maven section because of https://github.com/eclipse-pde/eclipse.pde/issues/675 -->
 			<unit id="com.google.guava.failureaccess" version="0.0.0" />
 
-			<unit id="javax.inject" version="0.0.0" />
-			<unit id="org.hamcrest.core" version="1.3.0.v20230721-0740" />
-			<unit id="org.hamcrest.library" version="1.3.0.v20180524-2246" />
-			<unit id="org.junit" version="4.13.2.v20230725-0701" />
+			<unit id="org.hamcrest.core" version="1.3.0.v20230809-1000" />
+			<unit id="org.hamcrest.library" version="1.3.0.v20230809-1000" />
+			<unit id="org.junit" version="4.13.2.v20230809-1000" />
 		</location>
 	</locations>
 	<targetJRE path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17" />

